tests/test_gatherdeferredlogger.cpp

```diff
@@ -80,14 +81,14 @@ void initLogger(std::ostringstream& log_stream) {
 
 BOOST_AUTO_TEST_CASE(NoMessages)
 {
-    auto cc = Dune::MPIHelper::getCollectiveCommunication();
+    const Opm::Parallel::Communication& cc = Dune::MPIHelper::getCollectiveCommunication();
```

Comment:
Seems like changes concerning `auto` to this file are totally unnecessary. Is there a reason not to rely on auto anymore?